dotnet/src/Functions/Functions.OpenApi/Model/RestApiOperation.cs

```diff
@@ -261,8 +262,30 @@ private Uri GetServerUrl(Uri? serverUrlOverride, Uri? apiHostUrl)
         }
         else
         {
+            Uri? serverUrl = null;
+            if (this.Server.Url is not null)
+            {
+                var url = this.Server.Url;
+                foreach (var variable in this.Server.Variables)
+                {
+                    if (arguments.TryGetValue(variable.Key, out object? value))
+                    {
+                        url = url.Replace($"{{{variable.Key}}}", value!.ToString());
+                    }
+                    else if (variable.Value.Default is not null)
+                    {
+                        url = url.Replace($"{{{variable.Key}}}", variable.Value.Default);
+                    }
+                    else
+                    {
+                        throw new KernelException($"No value provided for the '{variable.Key}' server variable of the operation - '{this.Id}'.");
+                    }
```

Comment:
If the server variable has the `enum` property set, the provided value should exist in the enum. i.e.

```yaml
url: http://localhost:{port}
description: The port
variables:
  port:
    enum: ["8443", "443"]
    default: 443
```
In the above snippet, a port value of `9000` should be rejected.

From [the spec](https://spec.openapis.org/oas/v3.0.3#fixed-fields-4):

> If the [enum](https://spec.openapis.org/oas/v3.0.3#serverVariableEnum) is defined, the value SHOULD exist in the enum’s values.

Reply:
Yep, I will add this validation